ONSdigital/florence
# Feature/health endpoint

### What

- Replaced old healthcheck with dp-healthcheck
- Add checkers for Vault (if we're using encryption), and S3
- Migrated S3 functionality from upload.go to dp-s3 repository.
- Use go modules (with go version 1.13)
- Refactored Makefile/CI so that they don't depend on $GOPATH
- Refactored Makefile generate step - we only create a single generated file (either prod or debug) to prevent duplicated functions and variables. Fixed test for the debug case (created testBindataFileInfo).
- replaced go-ns/log with log.go

### How to review

- Verify that the changes make sense
- Go tests should pass in both debug and prod generated files:
  - `make node-modules` `make generate-go-debug` `make test-go`
  - `make node-modules` `make generate-go-prod` `make test-go`
- try to run Florence: `make debug` and try the /health endpoint: `curl GET 'http://localhost:8081/health'`. If you run again with `ENCRYPTION_DISABLED=true`, then you should only see the S3 Check in the list.

### Who can review

Anyone
